### PR TITLE
[Kubernetes] Delete Legacy Ray Operator

### PR DESCRIPTION
**Pull request by DmitriGekhtman**
## Why are these changes needed?

Deletes the deprecated legacy Ray operator.

## Checks

- [ ] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: python/ray/autoscaler/_private/autoscaler.py [hunk 1]
```diff
@@ -357,13 +357,6 @@ def update(self):
         except Exception as e:
             self.prom_metrics.update_loop_exceptions.inc()
             logger.exception("StandardAutoscaler: Error during autoscaling.")
-            # Don't abort the autoscaler if the K8s API server is down.
-            # https://github.com/ray-project/ray/issues/12255
-            is_k8s_connection_error = self.config["provider"][
-                "type"
-            ] == "kubernetes" and isinstance(e, MaxRetryError)
-            if not is_k8s_connection_error:
-                self.num_failures += 1
```
Comment (DmitriGekhtman) [inline python/ray/autoscaler/_private/autoscaler.py:366]: Context: This code only affects "type" Kubernetes, which is being deleted.

For KubeRay, it's fine for the autoscaler to exit -- K8s will restart it.

Comment (DmitriGekhtman) [reply python/ray/autoscaler/_private/autoscaler.py:366]: (Though we could think about whether K8s crashlooping is ok or we want to handle things in a more controlled way.)

File: python/ray/autoscaler/_private/monitor.py [hunk 2]
```diff
@@ -138,7 +137,6 @@ def __init__(
         log_dir: str = None,
         prefix_cluster_info: bool = False,
         monitor_ip: Optional[str] = None,
-        stop_event: Optional[Event] = None,
```
Comment (DmitriGekhtman) [inline python/ray/autoscaler/_private/monitor.py:141]: Only used by legacy Ray Operator.

**Review by pcmoritz — APPROVED** [review]
Let's goooooo!

**Review by wuisawesome — APPROVED** [review]

**Review by edoakes — APPROVED** [review]

**Review by jjyao — APPROVED** [review]

**Comment by DmitriGekhtman** [comment]
Test failures are all pandas import errors, unrelated.

**Pull request merged**